### DISCUSSION TITLE
Monitoring WAN and InterVLANs

### DISCUSSION THREAD

**Discussion by eduncan911** [Security-Onion-Solutions/securityonion#3168] [Unsupported Versions]
Is SO capable of monitoring both WAN->LAN traffic as well as InterVLAN VLAN->VLAN traffic?

/tl;dr

New to SO, I'm trying to plan my server/VM allocation to set all of this up.  I've read up on Port SPANing/Mirroing which my L3 switch supports.

My 10Gbps InterVLAN are being routed via the L3 switch via ACLs, not the router, for hardware offloading.

I can place a 10 Gbps server between my router and L3 switch, for traffic blocking.  I'm concerned about latency though (as I game a lot at home).  So I may just port mirror for passive IDS.

But, how would I monitor InterVLAN traffic (VLAN to VLAN), as well WAN->LAN, on the same SO box?

Do I mirror multiple ports?  (one per VLAN?)  Does SO handle VLAN tagging, so I just need to mirror a Trunk line to SO?  Then, do I also mirror a 2nd WAN port to the same SO box?  (multiple NICs)

Can this VLAN trunk monitoring happen on the same SO install as placing it between WAN and L3 Switch for IPS?

I can setup a dedicated server with like 6 to 8 NICs in it (quad NIC cards).  Just trying to wrap my head around how SO ingests data.

**Comment by rwaight** [comment]
If you are looking to collect VLAN tags then you may have to add Packetbeat data.  I am looking to include VLAN tags and noticed in the [SO documentation](https://docs.securityonion.net/en/2.3/index.html) that [Zeek and Stenographer do not see VLAN tags](https://docs.securityonion.net/en/2.3/bpf.html?#vlan) (due to AF-PACKET).  I'm not sure if you would be able to add Packetbeat to an SO sensor, but otherwise you would need to configure Packetbeat to use `packetbeat.interfaces.type: pcap` and `packetbeat.interfaces.with_vlans: true`

The configuration for Packetbeat would need to include something similar to:
```
packetbeat.interfaces.device: eth0
packetbeat.interfaces.type: pcap
packetbeat.interfaces.with_vlans: true
```

You would also need to [configure flows to monitor network traffic](https://www.elastic.co/guide/en/beats/packetbeat/7.14/configuration-flows.html#configuration-flows) in the Packetbeat configuration.

Packetbeat doc links:  
* type:  https://www.elastic.co/guide/en/beats/packetbeat/7.14/configuration-interfaces.html#_type
* with_vlans:  https://www.elastic.co/guide/en/beats/packetbeat/7.14/configuration-interfaces.html#_with_vlans

**Reply by dougburks** [reply]
@rwaight To clarify, Zeek and Stenographer should correctly analyze traffic on a VLAN, it's just that they won’t log the actual VLAN tags. I've updated the documentation to reflect this:
https://docs.securityonion.net/en/2.3/bpf.html?#vlan

**Reply by eduncan911** [reply]
Thanks for the answer and doc updates!  I'm just about done with my HomeLab and then SO comes next.  So I'll try out these solutions and will make as answer.

The box I am using for SO has 6x 10G ports, so I should be able to port mirror everything to it.

**Reply by rwaight** [reply]
Thank you for the clarification and updating the documentation, @dougburks.  

In my use case, I am also looking to include a field with the vlan tag pre-populated (without any changes to parsing).  I was able to install and configure Packetbeat on the sensor as mentioned in https://github.com/Security-Onion-Solutions/securityonion/discussions/3168#discussioncomment-1288349 and can verify that the `so-beats*` index is populated with a `flow.vlan` field with the proper VLAN in the message.

**Reply by dougburks** [reply]
@rwaight Another option might be to use Suricata instead of Zeek as your metadata engine as Suricata should log VLAN tags correctly.
